test/pipeline_test_runner.cc

```diff
@@ -726,6 +726,19 @@ TEST_CASE("PerPhaseTest") { // NOLINT
         }
     }
 
+    {
+        // Non-mutating resolver.  Note that non-mutating resolver leaves GlobalState
+        // alone, but modifies trees with abandon, so we need to copy here.
+        vector<ast::ParsedFile> treesCopy;
+        for (auto &tree : trees) {
+            treesCopy.emplace_back(ast::ParsedFile{tree.tree.deepCopy(), tree.file});
+        }
+
+        move(resolver::Resolver::runIncrementalWithoutStateMutation(*gs, move(treesCopy)).result());
+        errorQueue->flushAllErrors(*gs);
+        errorCollector->drainErrors();
```

Comment:
We used to have `ENFORCE(!gs->hadCriticalError());` here in an earlier commit.

However, this `ENFORCE` would fail on `test/testdata/resolver/infinite.rb`.  This test is testing recursive dealiasing of constants and deliberately sets out to make Sorbet hit its internal limit of recursive dealiasing.  As a result, we hit:

https://github.com/sorbet/sorbet/blob/ebe537d200780ff64dab06eaca250b6fda579221/core/Symbols.cc#L1935-L1939

and this error is defined as `StrictLevel::Internal`:

https://github.com/sorbet/sorbet/blob/ebe537d200780ff64dab06eaca250b6fda579221/core/errors/internal.h#L8

such errors are considered "critical" by `Error` and wind up setting a flag on the error queue.

I cribbed this `ENFORCE` from jez's analogous testing for non-mutating namer, and naively considered that it would enforce that we hadn't `ENFORCE`'d or somesuch.  In retrospect, this was a silly thing for me to assume: `ENFORCE`s should just fire without respect to the errors raised on `GlobalState` (they do not interact with Sorbet's error reporting mechanism), and a generic test runner shouldn't assume anything about whether a particular testcase does or does not trigger critical errors.  (The one test we have checking `hadCriticalError` uses a testcase specifically designed to trigger such an error.)

I think "critical" is the wrong terminology to use for this, but that is a separate issue.

Long story short, I have deleted the `ENFORCE(!gs->hadCriticalError());`, partly in an effort to make tests pass, but partly because I think it really is the right thing to do.  Ideally folks will agree with the rationale above.  If not, we can discuss it later.